sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToStringConverter.cs

```diff
@@ -20,52 +21,73 @@ public async Task<string> ConvertAsync(ServiceBusReceivedMessage input, Cancella
             {
                 throw new ArgumentNullException(nameof(input));
             }
-            if (input.Body == null)
-            {
-                return null;
-            }
 
-            Stream stream = input.Body.ToStream();
-
-            TextReader reader = new StreamReader(stream, StrictEncodings.Utf8);
-            try
+            var body = input.GetRawAmqpMessage().Body;
+            if (body.BodyType == AmqpMessageBodyType.Data)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                try
+                if (input.Body == null)
                 {
-                    return await reader.ReadToEndAsync().ConfigureAwait(false);
-                }
-                catch (DecoderFallbackException)
-                {
-                    // we'll try again below
+                    return null;
                 }
 
-                // We may get here if the message is a string yet was DataContract-serialized when created. We'll
-                // try to deserialize it here using GetBody<string>(). This may fail as well, in which case we'll
-                // provide a decent error.
+                Stream stream = input.Body.ToStream();
```

Comment:
No using declarations?

Reply:
Why do we even go though stream and stream reader here? `input.Body.ToString()`?

Reply:
That would bypass the strict encoding instance used for reading.   I'm not familiar with the reason for it, but is that not significant here?

Reply:
Worst case StrictEncodings.Utf8.GetString(input.Body.toArray())

Reply:
I tried to update the existing code while making minimal changes. I will do another pass and fix things up.

Reply:
> Why do we even go though stream and stream reader here? input.Body.ToString()?

The point is to allow the DecoderFallbackException to be thrown and caught which signals that the data was DataContract serialized.

Reply:
But you are right - we can just do what you said - StrictEncodings.Utf8.GetString(input.Body.toArray())

Reply:
/cc @mathewc in case we are missing something here, but it seems like we can just use:
StrictEncodings.Utf8.GetString as opposed to stream/streamReader